### PR TITLE
relax pytest pin to have support for new `pytest==8.0.0`

### PR DESCRIPTION
**Pull request by valeriupredoi**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* ~[ ] Reset the build number to `0` (if the version changed)~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

New `pytest==8.0.0` has been out for a few days now, and since `pytest` is hard pinned to `<8.0.0` in the plugin, this leads to environments with latest `pytest` but `pytest-metadata ==2.0.0` which is ye olde and causes all manners of issues. See https://github.com/conda-forge/pytest-feedstock/issues/170 for a detailed env analysis. Also see [one of our CI fails](https://github.com/ESMValGroup/ESMValTool/actions/runs/7704516072/job/20996946958) and https://github.com/ESMValGroup/ESMValTool/issues/3513 for examoles of issues from the pytest=8/pytest-metadata=2 combo. Now, I don't normally go about updating pins in feedstocks without prior checking with upstream folks, but I see upstream don't have a hard upper limit on pytest in [their pyproject](https://github.com/pytest-dev/pytest-metadata/blob/master/pyproject.toml) so I took the liberty of updating it here, hope you don't mind (BTW I'd completely remove the upper pin, if upstream don't have one, but that's up to you to decide methinks). Cheers, fellas :beer: 

Fixes https://github.com/conda-forge/pytest-feedstock/issues/170
Fixes https://github.com/pytest-dev/pytest/issues/11893

### CODE REVIEW

**Comment by valeriupredoi** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ requirements:
 
   run:
     - python >=3.7
-    - pytest >=7.0.0,<8.0.0
+    - pytest >=7.0.0,<9.0.0
```
Comment (nicoddemus) [inline recipe/meta.yaml:28]: Actually upstream is now pinning only to >=7:

https://github.com/pytest-dev/pytest-metadata/blob/65ab3e4c3c714680e79fcf82b9795620e3f4640d/pyproject.toml#L42-L44

So I think we should follow suite here. :+1:

Comment (nicoddemus) [inline recipe/meta.yaml:28]:
```suggestion
    - pytest >=7.0.0
```

**Review by nicoddemus — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (valeriupredoi) [reply recipe/meta.yaml]: @nicoddemus cheers for the lightning speed lookup, mate! Completely agreed, lemme plop the change :+1:

Comment (valeriupredoi) [reply recipe/meta.yaml]: done via terminal, missed your GH codechange

Comment (nicoddemus) [reply recipe/meta.yaml]: No worries

**Comment by valeriupredoi** [comment]
sweet! Very many thanks @nicoddemus :beer:

**Pull request merged**